File: BurninWiper/MainActivity.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Preferences;
using Android.Support.Design.Widget;
using Android.Support.V7.App;
using Android.Support.V7.Widget;
using Android.Views;
using Android.Widget;
using Plugin.InAppBilling;
using System;
using static Android.Widget.AdapterView;

namespace BurninWiper
{
	[Activity(Label = "@string/app_name", Theme = "@style/AppTheme.NoActionBar", MainLauncher = true)]
	public class MainActivity : AppCompatActivity
	{
		public MainActivity()
		{
			if (falseflag)
			{
				var ignore = new FitWindowsLinearLayout(this);
				var ignore2 = new CoordinatorLayout(this);
			}
		}
		#region UI INITIALIZATIONS
		private Spinner TimerSpinner;
		private RadioGroup ModeGroup;
		private void InitializeControls()
		{
			TimerSpinner = FindViewById<Spinner>(Resource.Id.TimerSpinner);
			ModeGroup = FindViewById<RadioGroup>(Resource.Id.ModeRadioGroup);
			var startButton = FindViewById<Button>(Resource.Id.StartButton);
			var checkButton = FindViewById<Button>(Resource.Id.CheckButton);
			var contentMain = FindViewById<RelativeLayout>(Resource.Id.content_main);
			var donateButton = FindViewById<Button>(Resource.Id.DonateButton);
			LoadTimerItems();
			startButton.Click += StartButton_Click;
			checkButton.Click += CheckButton_Click;
			donateButton.Click += DonateButton_Click;

			LoadPreferences();

			TimerSpinner.ItemSelected += (_, __) =>
			 {
				 var spinnerTextView = TimerSpinner.GetChildAt(0) as TextView;
				 spinnerTextView?.SetTextSize(Android.Util.ComplexUnitType.Sp, 20);
				 spinnerTextView?.SetTextColor(Android.Content.Res.ColorStateList.ValueOf(Android.Graphics.Color.White));
			 };

		}

		static bool falseflag = false;
		protected override void OnCreate(Bundle savedInstanceState)
		{
			base.OnCreate(savedInstanceState);
			SetContentView(Resource.Layout.activity_main);
			InitializeControls();
		}
		private void LoadTimerItems()
		{
			var adapter = ArrayAdapter.CreateFromResource(
					this, Resource.Array.timer_array, Android.Resource.Layout.SimpleSpinnerItem);

			adapter.SetDropDownViewResource(Android.Resource.Layout.SimpleSpinnerDropDownItem);
			TimerSpinner.Adapter = adapter;
		}

		#endregion

		private void SavePreferences()
		{
			var prefs = PreferenceManager.GetDefaultSharedPreferences(this);
			
			var editor = prefs.Edit();
			editor.PutInt("ModeIndex", ModeGroup.CheckedRadioButtonId);
			editor.PutInt("TimerIndex", TimerSpinner.SelectedItemPosition);
			editor.Commit();
		}
		private void LoadPreferences()
		{
			var prefs = PreferenceManager.GetDefaultSharedPreferences(this);
			ModeGroup.ClearCheck();
			ModeGroup.Check(prefs.GetInt("ModeIndex", 0));
			TimerSpinner.SetSelection(prefs.GetInt("TimerIndex", 0));
		}

		private void DonateButton_Click(object sender, EventArgs e)
		{
			SavePreferences();
			var intent = new Intent(this, typeof(DonationActivity));
			StartActivity(intent);
		}


		private void CheckButton_Click(object sender, EventArgs e)
		{
			var prefs = PreferenceManager.GetDefaultSharedPreferences(this);
			var edit = prefs.Edit();
			edit.PutBoolean("IsChecking", true);
			edit.Commit();
			StartButton_Click(this, null);
		}


		private void StartButton_Click(object sender, EventArgs e)
		{
			SavePreferences();
			var intent = new Intent(this, typeof(ColorActivity));
			StartActivity(intent);
		}
	}
}

File: app/src/main/java/soheilkd/burninwiper/MainActivity.kt
package soheilkd.BurninWiper

import android.content.Intent
import android.os.Bundle
import android.widget.ArrayAdapter
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)
		setSupportActionBar(toolbar)
		loadTimerItems()
		loadPreferences()

		main_startButton.setOnClickListener { run { start() } }
		main_checkButton.setOnClickListener { run { start(true) } }
		main_donateButton.setOnClickListener { run { openDonate() } }
	}

	override fun onDestroy(){
		savePreferences()
		super.onDestroy()
	}

	private fun savePreferences() {
		val prefs = PreferenceManager.getDefaultSharedPreferences(this)

		val editor = prefs.edit()
		editor.putInt("ModeId", main_radioGroup.checkedRadioButtonId)
		editor.putInt("TimerIndex", main_timeSpinner.selectedItemPosition)
		editor.apply()
	}

	private fun loadPreferences() {
		val prefs = PreferenceManager.getDefaultSharedPreferences(this)
		main_radioGroup.clearCheck()
		main_radioGroup.check(prefs.getInt("ModeId", R.id.main_radio1))
		main_timeSpinner.setSelection(prefs.getInt("TimerIndex", 0))
	}

	private fun loadTimerItems() {
		val timerItems =
			arrayOf(getString(R.string.ui_timerNone), "5m", "10m", "20m", "30m", "1h", "2h", "3h")
		val adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, timerItems)
		adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
		main_timeSpinner.adapter = adapter
	}

	private fun start(isChecking: Boolean = false) {
		savePreferences()
		val intent =
			when (main_radioGroup.checkedRadioButtonId) {
				R.id.main_radio4 -> Intent(this, AnimActivity::class.java) //aka animation mode
				else -> Intent(this, ColorActivity::class.java) //aka other modes
			}
		intent.putExtra("TimerIndex", main_timeSpinner.selectedItemPosition)
		intent.putExtra("ColorMode", main_radioGroup.checkedRadioButtonId)
		intent.putExtra("IsChecking", isChecking)
		intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT)
		startActivity(intent)
	}

	private fun openDonate() {
		savePreferences()
		val intent = Intent(this, DonationActivity::class.java)
		startActivity(intent)
	}
}
